go/vt/mysqlctl/mysqld.go

```diff
@@ -107,7 +109,19 @@ func NewMysqld(dbcfgs *dbconfigs.DBConfigs) *Mysqld {
 	result.appPool = dbconnpool.NewConnectionPool("AppConnPool", *appPoolSize, *appIdleTimeout, *poolDynamicHostnameResolution)
 	result.appPool.Open(dbcfgs.AppWithDB(), appMysqlStats)
 
-	return result
+	var err error
+	result.flavor, err = result.detectFlavor()
```

Comment:
Is there any reason `detectFlavor` needs to be a separate function from `detectVersion`? We seem to only ever call them here, back-to-back. Can we just have `detectVersion` fill in both flavor and version whenever possible? Both functions do basically the same thing (look at MYSQL_FLAVOR and call `getVersionString`).